.pre-commit-config.yaml

```
@@ -32,6 +32,12 @@ repos:
       - id: fix-encoding-pragma
         args: [--remove]
         exclude: ^conda/auxlib/
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.38.2
+    hooks:
+      - id: pyupgrade
+        args: ["--py37-plus", "--keep-runtime-typing"]
```

I don't think we need to preserve the typing?

```suggestion
        args: ["--py37-plus"]
```

---

By removing `--keep-runtime-typing` we'll get types like `str | None` or `list[str]` that aren't parsed at runtime. It is annoying when pyupgrade edits your carefully typed types but it is more modern.